Source/ASCollectionView.mm

```diff
@@ -2015,7 +2015,22 @@ - (ASCellNodeBlock)dataController:(ASDataController *)dataController supplementa
     block = ^{ return cell; };
   }
 
-  return block;
+  // Wrap the node block
+  __weak __typeof__(self) weakSelf = self;
+  return ^{
+    __typeof__(self) strongSelf = weakSelf;
+    ASCellNode *node = (block ? block() : cell);
```

Comment:
Isn't block is guaranteed to be non-nil after [this line](https://github.com/TextureGroup/Texture/pull/1265/files#diff-8b82b1556d1293f6f9789beb236c9271R2015)? Maybe remove that line to be consistent with the implementation of `dataController:nodeBlockAtIndexPath:`.

Reply:
Concur with @nguyenhuy. The block should not be nil in here and the returning block is retaining the block too.

Reply:
I thought you had to `copy` a block to actually retain it

Reply:
nm I'm like stuck in 2010 😛